app/scripts/setup.sh

```diff
@@ -161,16 +161,19 @@ DEV_VERSION="dev"
 if [ ! -d ${install}/.git ]
 then
     CURRENT_TIP="unknown"
+    LATEST_TAG_NAME="unknown"
 else
     cd "${install}"
     CURRENT_TIP=$(git rev-parse HEAD)
     LATEST_TAG=$(git describe --abbrev=0 --tags | xargs -I {} git rev-list -n 1 {})
+    LATEST_TAG_NAME=$(git describe --abbrev=0 --tags)
     if [ "${CURRENT_TIP}" = "${LATEST_TAG}" ]
     then
         DEV_VERSION=""
     fi
 fi
 echo "${CURRENT_TIP:0:7}${DEV_VERSION}" > "${install}/cache/version.txt"
+echo "${LATEST_TAG_NAME}" | tr -d '\n' > "${install}/cache/tag.txt"
```

Comment:
I'd keep the new line here, and deal with it when reading the file like we do for `version.txt`.

```
echo "${LATEST_TAG_NAME}" > "${install}/cache/tag.txt"
```

Reply:
We really don’t want new lines in this file, so it’s not obvious to me why it’s better to take the risk of forgetting removing it on the other end

Reply:
Because if for any reason someone edits the file manually?

Reply:
I think it's mostly a matter of leaving the command output alone. If you prefer to remove the new line here (note that your code is already doing it twice though), I'd update the code for the version to follow the same logic.